File: 백준/Silver/9465. 스티커/스티커.java
import java.io.*;
import java.util.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st;
        StringBuilder sb = new StringBuilder();

        int t = Integer.parseInt(br.readLine());

        for (int i = 0; i < t; i++) {
            int n = Integer.parseInt(br.readLine());
            int[][] dp = new int[2][n + 1];
            int[][] nums = new int[2][n];

            for (int j = 0; j < 2; j++) {
                st = new StringTokenizer(br.readLine());
                for (int k = 0; k < n; k++) {
                    nums[j][k] = Integer.parseInt(st.nextToken());
                }
            }

            dp[0][1] = nums[0][0];
            dp[1][1] = nums[1][0];

            for (int j = 2; j < n+1; j++) {
                int up = nums[0][j-1];
                int down = nums[1][j-1];

                dp[0][j] = up + Math.max(dp[1][j-1], dp[1][j-2]);
                dp[1][j] = down + Math.max(dp[0][j-1], dp[0][j-2]);
            }
            sb.append(Math.max(dp[0][n], dp[1][n])).append("\n");
        }

        System.out.println(sb);
    }
}

File: 백준/Silver/9465. 스티커/스티커.py
# 스티커
import sys
input = sys.stdin.readline

t = int(input())
for i in range(t):
    n = int(input())
    # dp[i][j] => i행 j열 까지 스티커를 떼어냈을 때 최대값
    dp = [[0] * (n+1) for _ in range(2)]
    nums = []

    for j in range(2):
        nums.append(list(map(int,input().split())))

    dp[0][1] = nums[0][0]
    dp[1][1] = nums[1][0]
    
    for i in range(2, n+1):
        up = nums[0][i-1]
        down = nums[1][i-1]
        
        dp[0][i] = up + max(dp[1][i-1], dp[1][i-2])
        dp[1][i] = down + max(dp[0][i-1], dp[0][i-2])
        
    print(max(dp[0][n], dp[1][n]))
